test/wait-task/pkg/reconciler/reconciler.go

```diff
@@ -67,27 +74,67 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, r *v1alpha1.Run) kreconc
 		return nil
 	}
 
-	dur, err := time.ParseDuration(expr.Value.StringVal)
-	if err != nil {
-		r.Status.MarkRunFailed("InvalidDuration", "The duration param was invalid: %v", err)
+	// Skip if the Run is cancelled.
+	if r.IsCancelled() {
+		logger.Infof("The Custom Task Run %v has been cancelled", r.GetName())
+		// Check if the Run has been cancelled by its Parent PipelineRun
+		if r.Spec.Status == v1alpha1.RunSpecStatusCancelled && r.Spec.StatusMessage == v1alpha1.RunCancelledByPipelineMsg {
+			now := metav1.Now()
+			r.Status.CompletionTime = &now
+			r.Status.MarkRunFailed("ParentPipelineRunTimedOut", WaitTaskCancelledByPipelineRunTimeoutMsg)
+		}
```

Comment:
I think the status message should just be propagated to the run status-- runs don't need to know about pipelineruns

Reply:
SG, updated